third_party/move/move-compiler/src/parser/syntax.rs

```diff
@@ -1901,9 +1906,35 @@ fn at_start_of_exp(context: &mut Context) -> bool {
     )
 }
 
+// Parse the rest of a lambda expression, after already processing any capture designator (move/copy/&).
+fn parse_lambda(
+    context: &mut Context,
+    start_loc: usize,
+    capture_kind: LambdaCaptureKind,
+    token: Tok,
+) -> Result<Exp_, Box<Diagnostic>> {
+    let bindings = if token == Tok::Pipe {
+        parse_lambda_bind_list(context)?
+    } else {
+        // token is Tok::PipePipe, i.e., empty bind list in this context.
+        consume_token(context.tokens, Tok::PipePipe)?;
+        spanned(context.tokens.file_hash(), start_loc, start_loc + 1, vec![])
+    };
+    let body = Box::new(parse_exp(context)?);
+    let abilities_start = context.tokens.start_loc();
+    let abilities = parse_abilities(context)?;
```

Comment:
The abilities of a function type should use `|T|R with A` and not `:`. It looks strange to have two colons in an argument declaration, as in `f(x: |T|:copy`. Thats also the reason why it is not `|T|:R`.